cmd/prometheus/main.go

```diff
@@ -85,6 +85,10 @@ func init() {
 	if err != nil {
 		panic(err)
 	}
+
+	// Define fixed length millisecond timestamp format.
+	log.DefaultTimestamp = log.TimestampFormat(time.Now, "2006-01-02T15:04:05.000Z07:00")
+	log.DefaultTimestampUTC = log.TimestampFormat(func() time.Time { return time.Now().UTC() }, "2006-01-02T15:04:05.000Z07:00")
```

Comment:
How about putting this into the `promlog` package (that's used here: https://github.com/prometheus/prometheus/blob/4bbd72ec877a8dacbeff015e6aa87b37c3984f4d/cmd/prometheus/main.go#L296). That way our other binaries that use `promlog` also get this benefit. Also, is there a non-global way of doing this (e.g. only returning it as a property of the returned logger)?

Reply:
Yes, see https://github.com/prometheus/common/pull/172

Reply:
Oh whoops, I totally missed that there was previous conversation on this PR. Sorry!